src/main.rs

```diff
@@ -59,11 +89,55 @@ fn main() {
                 .expect("Error dumping entries");
         }
         InfluxRpcCompare::DumpCalls(dump) => {
-            dump_calls::DumpCalls::new(dump.path)
-                .dump(&mut stdout())
-                .expect("Error dumping calls");
+            if matches!(dump.format, CallFormat::Binary) && dump.output_path.is_none() {
+                eprintln!("output path required");
+                return;
+            }
+
+            let mut dc = dump_calls::DumpCalls::new(dump.input_path);
+            let calls = match dc.process() {
+                Ok(calls) => calls,
+                Err(e) => {
+                    eprintln!("{}", e);
+                    return;
+                }
+            };
+
+            // Filter offset calls out
+            let calls = calls.filter_offset_calls();
+            println!("Filtered offset calls. {:?} calls remaining", calls.len());
+
+            // Filter by org_id
+            let calls = if dump.org_filter.is_empty() {
+                calls
+            } else {
+                let org_filter = dump.org_filter.as_str();
+                let calls = calls.filter_by_org_id(org_filter);
+                println!(
+                    "Filtered calls not for org id {}. {:?} calls remaining",
+                    org_filter,
+                    calls.len()
+                );
+                calls
+            };
+
+            let res = match dump.format {
+                CallFormat::Pretty => dc.write_calls_pretty(calls, &mut stdout()),
+                CallFormat::Binary => dc.write_calls_binary(
+                    calls,
+                    dump.output_path
```

Comment:
To parse into a PathBuf only to convert back to &str seems a little funky, perhaps `write_calls_binary` should take `impl Into<Path>` or something